backend/common/repo.py

```diff
@@ -214,19 +214,22 @@ def is_flat(self) -> bool:
 
     def verify_packages_index(self) -> bool:
         """
-        Verify Packages index against all listed checksum algorithms.
+        Verify Packages index with the best available checksum algorithm.
 
-        :param name: name of the packages index
         :return: result (boolean)
         """
         name, data = self.get_pkg_index_raw()
         entry = self.get_release_index().get(name)
-        for algorithm in ["md5", "sha1", "sha256"]:
-            if entry is None:
-                res = False
-            else:
-                res = getattr(hashlib, algorithm)(data).hexdigest() == getattr(entry.checksum, algorithm)
-            if not res:
+        if entry is None:
+            return False
+
+        result = False
+        for algorithm in ("sha512", "sha384", "sha256", "sha1", "md5"):
```

Comment:
You need to adapt the parser as well. Currently only "sha256", "sha1", "md5" are supported.
Maybe currently debian do not use better algorithms, but sooner or later they will. So why not support them already.